backend/controllers/error_controller.py

```diff
@@ -0,0 +1,142 @@
+from data_reader import get_data
+from numpy import percentile, asarray
+
+error_importance_by_files = {}
+
+
+def get_error(segment1, segment2):
+    """
+
+    This function takes two segments as parameters, and starts iterating over their dates and
+    inventory_volumes arrays in order to calculate the error. Since the segments can have information
+    for different dates, the iteration is done by constantly aligning the next date for which the error
+    is going to be calculated. The starting position of the error calculation is the first date
+    that has produced a difference inn inventory_volumes as it is considered the first prediction.
+    The error is calculated as (value2 - value1)/value2.
+
+    :param segment1: data_pb2.SegmentedData This is the first segment
+    :param segment2: data_pb2.SegmentedData This is the second segment
+    :return: data_pb2.SegmentedDataError This is the object containing all error information
+    """
+    error = data_pb2.SegmentedDataError(country=segment1.country, device=segment1.device)
+    index1, index2 = get_index_with_first_difference(segment1, segment2)
+    while index1 != -1:
+        value1 = segment1.inventory_volumes[index1]
+        value2 = segment2.inventory_volumes[index2]
+        if value1 != 0:
+            current_error = (value2 - value1) / value1
+            error.error.append(current_error)
+            error.dates.append(segment1.dates[index1])
+        index1 = index1 + 1
+        index2 = index2 + 1
+        index1, index2 = get_next_same_date_indexes(segment1.dates, segment2.dates, index1, index2)
+    error.min = min(error.error)
+    error.max = max(error.error)
+    quartiles = percentile(asarray(error.error), [25, 50, 75])
+    error.first_quartile = quartiles[0]
+    error.median = quartiles[1]
+    error.third_quartile = quartiles[2]
+    return error
+
+
+def get_index_with_first_difference(segment1, segment2):
+    """
+
+    This method gets two segments as parameters and starts iterating over their inventory_volumes
+    in order to find the indices of the first date for which the volumes are different.
+    Note that this method returns two indices as the same date may not correspond to the same index in the two arrays
+
+    :param segment1: data_pb2.SegmentedData This is the first segment
+    :param segment2: data_pb2.SegmentedData This is the second segment
+    :return: int, int Two indices that mark the position of the first difference between the two arrays
+    """
+    index1, index2 = get_next_same_date_indexes(segment1.dates, segment2.dates, 0, 0)
+    if index1 == -1:
+        return -1, -1
+    while segment1.inventory_volumes[index1] == segment2.inventory_volumes[index2]:
+        index1, index2 = get_next_same_date_indexes(segment1.dates, segment2.dates, index1, index2)
+        if index1 == -1:
+            return -1, -1
+    return index1, index2
+
+
+def get_next_same_date_indexes(dates1, dates2, index1, index2):
```

Comment:
The name is a bit confusing, as it does not necessarily return the next index if the current one pair is of the same date. Maybe "first", but that's also confusing...

Reply:
renamed to "get_first_same_date_indexes_in_sublists"